learn/app-development/widgets/datalive/dynamic-form.md

```diff
@@ -0,0 +1,155 @@
+[on-beforerender event](#modifying-dynamic-form-metadata) on the Form to modify the data accordingly.
+:::
+
+### Importing Service
+
+Go to Web Services. Click **Import Web Service** to import a new web service. To know how to import REST Web Service, see [Import REST Services](https://docs.wavemaker.com/learn/app-development/services/web-services/rest-services#test-rest-service-api).
+
+### Creating Variable
+
+Create a variable `MetadataVariable` to fetch the metadata. To know how to create a variable, see [Creating Service Variable](https://docs.wavemaker.com/learn/app-development/variables/web-service#how-to-create-a-service-variable).
+
+The created variable should be used in the **Markup** tab to [bind the metadata with the Form](#binding-variable-with-form).  
+
+[![](/learn/assets/variable-markup-dynamicform.png)](/learn/assets/variable-markup-dynamicform.png)
+
+### Creating Dynamic Form
+
+1. Go to pages and create a new page. To create a new page, see [Creating Page](https://docs.wavemaker.com/learn/app-development/ui-design/page-creation).
+
+#### Binding Variable with Form
+
+1. Go to the Markup tab and enter the following code snippet to create a Dynamic Form. Add metadata property in the markup with the created variable. For example, the accepted value for metadata is `metadata="bind:Variables.<Variable Name>.dataSet"`.
+
+```markup
+
+<wm-form errormessage="" captionposition="top" title="Form" enctype="application/x-www-form-urlencoded" method="post" metadata="bind:Variables.createMetadata.dataSet" captionalign="left" name="form1" on-beforerender="form1BeforeRender($metadata, widget)">
+                <wm-form-action key="reset" class="form-reset btn-secondary" iconclass="wi wi-refresh" display-name="Reset" type="reset"></wm-form-action>
+                <wm-form-action key="save" class="form-save btn-success" iconclass="wi wi-save" display-name="Save" type="submit"></wm-form-action>
+            </wm-form>
+
+```
+
+[![](/learn/assets/add-formcode-dynamicform.png)](/learn/assets/add-formcode-dynamicform.png)
+
+2. Click **Save** to save the changes in the Markup tab. Click Preview to view the created Dynamic Form.
+
+[![](/learn/assets/confirm-formcode-dynamicform.png)](/learn/assets/confirm-formcode-dynamicform.png)
+
+#### Modifying Dynamic Form Metadata
+
+1. If the Form fields do not render as expected, use the **on-beforerender** event to modify the provided field details into WaveMaker accepted format. For example, use the following format `on-beforerender="<Form_Widget_Name>Beforerender($metadata, $isolateScope)"` where `$isolateScope` is the accepted widget type.
+
+[![](/learn/assets/add-formcode-onbeforerender-dynamicform.png)](/learn/assets/add-formcode-onbeforerender-dynamicform.png)
+
+You can preview the Dynamic Form.
+
+## Create Business User Form
+
+To create Business User (BU) Form, you can create Static Form only in cases when BU wants to add or edit fields through WaveMaker.
```

Comment:
These steps are not required here. We can just summarize that "the metadata can be configured by a Business User through another static form." Then link to the static form creation.